src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+            }
+        }
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<4>>
+{
+    template<typename Weights, typename WeightVal>
+    static void set_weights(Weights& weights, WeightVal const& weight_val)
+    {
+        for (uint8_t i = 0; i < 4; i++)
+            weights[i] = weight_val;
+    }
+
+    template<typename Deltas, typename DeltaVal>
+    static void set_deltas(Deltas& deltas, DeltaVal const& delta_val)
+    {
+        deltas[0] = {-delta_val, -delta_val};
+        deltas[1] = {-delta_val, +delta_val};
+        deltas[2] = {+delta_val, -delta_val};
+        deltas[3] = {+delta_val, +delta_val};
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Cross>
+{
+    template<typename Weights, typename WeightVals>
+    static void set_weights(Weights& weights, WeightVals const& weight_vals)
+    {
+        for (uint8_t i = 0; i < 2; i++)
+            weights[i] = weight_vals[1];
+
+        for (uint8_t i = 2; i < 6; i++) // inner square
+            weights[i] = weight_vals[0];
+
+        for (uint8_t i = 6; i < 8; i++)
+            weights[i] = weight_vals[1];
+    }
+
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals)
+    {
+        deltas[0] = {-delta_vals[1], -delta_vals[1]};
+        deltas[1] = {-delta_vals[1], +delta_vals[1]};
+
+        deltas[2] = {-delta_vals[0], -delta_vals[0]}; // inner square
+        deltas[3] = {-delta_vals[0], +delta_vals[0]};
+        deltas[4] = {+delta_vals[0], -delta_vals[0]};
+        deltas[5] = {+delta_vals[0], +delta_vals[0]};
+
+        deltas[6] = {+delta_vals[1], -delta_vals[1]};
+        deltas[7] = {+delta_vals[1], +delta_vals[1]};
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Square>
+{
+    template<typename Weights, typename WeightVals>
+    static void set_weights(Weights& weights, WeightVals const& weight_vals)
+    {
+        for (uint8_t i = 0; i < 8; i += 2)
+            weights[i] = weight_vals[1];
+
+        for (uint8_t i = 1; i < 8; i += 2)
+            weights[i] = weight_vals[0];
+    }
+
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals)
+    {
+        deltas[0] = {-delta_vals[1], -delta_vals[1]};
+        deltas[1] = {0.0f, -delta_vals[0]};
+        deltas[2] = {+delta_vals[1], -delta_vals[1]};
+
+        deltas[3] = {-delta_vals[0], 0.0f};
+        deltas[4] = {+delta_vals[0], 0.0f};
+
+        deltas[5] = {-delta_vals[1], delta_vals[1]};
+        deltas[6] = {0.0f, delta_vals[0]};
+        deltas[7] = {+delta_vals[1], delta_vals[1]};
+    }
```

Comment:
I agree: this one is `PINK`  `PURPLE`, and is the only one we should keep.

Reply:
ok thank, this makes things simpler

Reply:
I think the setting of weights is wrong on this and should be 
![image](https://user-images.githubusercontent.com/30491/82900156-63e54980-9f5c-11ea-8e3f-d1e699c4a7a2.png)